gpkit/model.py

```diff
@@ -1,802 +0,0 @@
-            else:
-                posynomials.append(constraint)
-        return posynomials
-
-    @property
-    def beforesubs(self):
-        "Get this Model's NomialData before any substitutuions"
-        return NomialData.fromnomials(self.signomials)
-
-    @property
-    def allsubs(self):
-        "All substitutions currently in the Model."
-        subs = self.beforesubs.values
-        subs.update(self.substitutions)
-        return subs
-
-    @property
-    def signomials_et_al(self):
-        "Get signomials, beforesubs, allsubs in one pass; applies VarKey subs."
-        signomials = self.signomials
-        # don't use self.beforesubs here to avoid re-computing self.signomials
-        beforesubs = NomialData.fromnomials(signomials)
-        allsubs = beforesubs.values
-        allsubs.update(self.substitutions)
-        varkeysubs = {vk: nvk for vk, nvk in allsubs.items()
-                      if isinstance(nvk, VarKey)}
-        if varkeysubs:
-            for nvk in varkeysubs.values():
-                if "value" in nvk.descr:
-                    allsubs.update({nvk: nvk.descr["value"]})
-            beforesubs = beforesubs.sub(varkeysubs, require_positive=False)
-            beforesubs.varkeysubs = varkeysubs
-            signomials = [s.sub(varkeysubs, require_positive=False)
-                          for s in signomials]
-        return signomials, beforesubs, allsubs
-
-    @property
-    def constants(self):
-        "All constants (non-sweep substitutions) currently in the Model."
-        _, beforesubs, allsubs = self.signomials_et_al
-        return get_constants(beforesubs, allsubs)
-
-    def zero_lower_unbounded_variables(self):
-        "Recursively substitutes 0 for variables that lack a lower bound"
-        zeros = True
-        while zeros:
-            bounds = self.gp(verbosity=0).missingbounds
-            zeros = {var: 0 for var, bound in bounds.items()
-                     if bound == "lower"}
-            self.substitutions.update(zeros)
-
-    def solve(self, solver=None, verbosity=2, skipsweepfailures=False,
-              *args, **kwargs):
-        """Forms a mathematical program and attempts to solve it.
-
-        Arguments
-        ---------
-        solver : string or function (optional)
-            If None, uses the default solver found in installation.
-
-        verbosity : int (optional)
-            If greater than 0 prints runtime messages.
-            Is decremented by one and then passed to programs.
-
-        skipsweepfailures : bool (optional)
-            If True, when a solve errors during a sweep, skip it.
-
-        *args, **kwargs : Passed to solver
-
-        Returns
-        -------
-        sol : SolutionArray
-            See the SolutionArray documentation for details.
-
-        Raises
-        ------
-        ValueError if called on a model with Signomials.
-        RuntimeWarning if an error occurs in solving or parsing the solution.
-        """
```

Comment:
we lost this nice docstring too -- can parts of it be retained and put somewhere?